### PR TITLE
[docs] Create newsletter_september2023.md

### PR DESCRIPTION
**Pull request by MaximilianLombardo**
NewsLetter Census API September 2023 - covering:

top two points on this issue:
https://github.com/chanzuckerberg/cellxgene-census/pull/716

and the top two points on this issue:
https://github.com/chanzuckerberg/cellxgene-census/pull/612

[EDIT BY @pablo-gar] addresses: #726

### CODE REVIEW

File: docs/articles/2023/newsletter_september2023.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+## Schema Changes in V1.1.0
+
+### Added `dataset_version_id` to Census Table
+
+The new field `dataset_version_id` has been introduced in `census_obj["census_info"]["datasets"]` to facilitate dataset versioning and management.
+
+### Added New Library-Size Normalized Layer in X["normalized"]
+
+We've introduced a library-size normalized layer in X["normalized"]. A small sigma value has been consistently added to each position, ensuring that no explicit zeros are recorded.
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:18]: Please include a sentence that reflects the following message.

> The normalized layer is effectively a division for all values in the count matrix by the corresponding row  (cell) sum of counts. As such, numerical precision of a division can always be impacted, and in addition we introduced a reduction in precision to reduce the disk size required to save these values. As a consequence of that and contrary to the expectation, the row sums of the normalized layer is not guaranteed to be 1.0, but rather slightly lower or greater by a few decimal points.
>
> A small sigma value [...]

Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:18]:
```suggestion
We've introduced a library-size normalized layer for the RNA measurements of both the human and mouse experiments under `X["normalized"]`. [TEXT FROM COMMENT ABOVE]. A small sigma value has been added during the calculation of the normalized layer to ensure that all non-zero values from the count matrix are also non-zero values in the normalized layer.
```

File: docs/articles/2023/newsletter_september2023.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+These features are currently experimental and exclusive to the "latest" version of Census. We invite your feedback as you explore these novel functionalities.
+
+## Schema Changes in V1.1.0
+
+### Added `dataset_version_id` to Census Table
+
+The new field `dataset_version_id` has been introduced in `census_obj["census_info"]["datasets"]` to facilitate dataset versioning and management.
+
+### Added New Library-Size Normalized Layer in X["normalized"]
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:16]:
```suggestion
### Added new library-size normalized layer
```

File: docs/articles/2023/newsletter_september2023.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+*Published: 25 August 2023*  
+*By: [Author Name](mailto:author1@chanzuckerberg.com), [Co-Author](mailto:author2@chanzuckerberg.com)*
+
+The Census team is thrilled to announce the introduction of Census schema V1.1.0. This version offers enhanced metadata and API functionalities, tailored to empower your single-cell research.
+
+These features are currently experimental and exclusive to the "latest" version of Census. We invite your feedback as you explore these novel functionalities.
+
+## Schema Changes in V1.1.0
+
+### Added `dataset_version_id` to Census Table
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:12]:
```suggestion
### Added `dataset_version_id` to the datasets data frame
```

File: docs/articles/2023/newsletter_september2023.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+The new field `dataset_version_id` has been introduced in `census_obj["census_info"]["datasets"]` to facilitate dataset versioning and management.
+
+### Added New Library-Size Normalized Layer in X["normalized"]
+
+We've introduced a library-size normalized layer in X["normalized"]. A small sigma value has been consistently added to each position, ensuring that no explicit zeros are recorded.
+
+### Enhanced Metadata Fields in `ms["RNA"].var` DataFrame
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:20]:
```suggestion
### Enhanced gene metadata
```

File: docs/articles/2023/newsletter_september2023.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+The `ms["RNA"].var` DataFrame has been enriched with two new metadata fields: `nnz` and `n_measured_obs`, which provide a count of non-zero values and "measured" cells, respectively.
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:22]:
```suggestion
The `ms["RNA"].var` data frame for both the human and mouse experiments, has been enriched with two new metadata fields: `nnz` and `n_measured_obs`, which provide a count of non-zero values and "measured" cells, respectively.
```

File: docs/articles/2023/newsletter_september2023.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+### Enhanced Metadata Fields in `ms["RNA"].obs` DataFrame
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:24]:
```suggestion
### Enhanced cell metadata
```

File: docs/articles/2023/newsletter_september2023.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+The `obs` DataFrame is now augmented with the following new metadata, allowing users to forego common calculations used in early data pre-processing:
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:26]:
```suggestion
The `obs` data frame for both the human and mouse experiments is now augmented with the following new metadata, allowing users to forego common calculations used in early data pre-processing. For each cell:
```

File: docs/articles/2023/newsletter_september2023.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+- `raw_sum`: Represents the count sum derived from X["raw"]
+- `nnz`: Enumerates the number of non-zero (nnz) values
+- `raw_mean`: Provides the average count of nnz values
+- `raw_variance`: Indicates the count variance of nnz values
+- `n_measured_vars`: Enumerates the "measured" genes, determined by the sum of the presence matrix
+
+## How to Use the New Features
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:34]:
```suggestion
## How to use the new features
```

File: docs/articles/2023/newsletter_september2023.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+### Exporting Normalized Data
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:36]:
```suggestion
### Exporting normalized data
```

File: docs/articles/2023/newsletter_september2023.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+
+Normalized data can be exported into AnnData with the following code:
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:38]:
```suggestion
Normalized data can be exported into AnnData specifying the `X_name = "normalized"` argument of the `cellxgene.get_anndata()` method:
```

File: docs/articles/2023/newsletter_september2023.md [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+
+```python
+import cellxgene_census
+
+with cellxgene_census.open_soma() as census:
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:43]: This code is not going to work because the latest version of the Census data is not being opened.
```suggestion
with cellxgene_census.open_soma(census_version = "latest") as census:
```

File: docs/articles/2023/newsletter_september2023.md [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+        column_names = {"obs": ["assay", "cell_type", "tissue", "tissue_general", "suspension_type", "disease"]},
+        X_name = "normalized" # Specificy the normalized layer for this query
+        
+    )
+    
+    print(adata)
+
+#Adata with normalized layer
+```
+
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:58]: Please add SingleCellExperiment and Seurat examples

File: docs/articles/2023/newsletter_september2023.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+### Accessing Library-Size Normalized Data Layer via TileDB-SOMA 
+
+For memory-efficient data retrieval, you can use TileDB-SOMA as outlined below:
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:61]: Please explain a bit more what's the workflow is here:

1. Perform an axis query with the specified filter.
2. Obtain an iterator of the X matrix and specifically the normalized layer

File: docs/articles/2023/newsletter_september2023.md [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+        var_value_filter = "feature_id in ['ENSG00000161798', 'ENSG00000188229']",
+        obs_value_filter = "nnz > '5' and sex == 'female' and cell_type in ['microglial cell', 'neuron']", 
+        column_names = {"obs": ["assay", "cell_type", "tissue", "tissue_general", "suspension_type", "disease"]},
+    )
+
+    print(adata)
+
+#adata with filtered cells
+```
+
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:116]: Please include equivalent examples for Seurat and SingleCellExperiment

File: docs/articles/2023/newsletter_september2023.md [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+---
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:117]:
```suggestion
## Help us improve these data additions
```

File: docs/articles/2023/newsletter_september2023.md [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+
+We encourage you to engage with these new features in the Census API and share your feedback. This input is invaluable for the ongoing enhancement of the Census project.
+
+For further information on the new library-size normalized layer, please reach out to us at [cellxgene@chanzuckerberg.com](cellxgene@chanzuckerberg.com). To report issues or for additional feedback, refer to our [Census GitHub repository](https://github.com/chanzuckerberg/cellxgene-census/issues).
+
+---
```
Comment (pablo-gar) [inline docs/articles/2023/newsletter_september2023.md:123]:
```suggestion
```

**Review by pablo-gar — CHANGES_REQUESTED** [review]
In line suggestions and comments.

Additionally:

1. Please remove the camel case of all headers.
2. Please run and test the code locally, most of the code chunks are going to produce errors as they are due to the `open_soma()` call lacking the `census_version = "latest" argument.
3. Please rename the file to adhere to the publication editorial guidelines.

File: docs/articles/2023/newsletter_september2023.md (hunk 1 above)
Comment (MaximilianLombardo) [reply docs/articles/2023/newsletter_september2023.md:18]:
```suggestion
We've introduced a library-size normalized layer for the RNA measurements of both the human and mouse experiments under `X["normalized"]`. The normalized layer divides each value in the count matrix by its corresponding row sum (i.e. size normalization), leading to potential numerical precision issues and slight deviations from the expected row sum of 1.0. A small sigma value has been added during the calculation of the normalized layer to ensure that all non-zero values from the count matrix are also non-zero values in the normalized layer.
```

File: docs/articles/2023/newsletter_september2023.md [hunk 17]
```diff
@@ -0,0 +1,123 @@
+# Introducing Schema V1.1.0: Key Updates in Census API and Data Layer
+
+*Published: 25 August 2023*  
+*By: [Author Name](mailto:author1@chanzuckerberg.com), [Co-Author](mailto:author2@chanzuckerberg.com)*
```
[the remark this answers is not in the document]
Comment (MaximilianLombardo) [reply docs/articles/2023/newsletter_september2023.md:4]:
```suggestion
*By: [Maximilian Lombardo](mailto:mlombardo@chanzuckerberg.com)*
```

**Comment by ebezzi** [comment]
@MaximilianLombardo and @pablo-gar - is this ready to go?

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 18]
```diff
@@ -0,0 +1,225 @@
+# Introducing a normalized layer and pre-calculated cell and gene statistics in Census
+
+*Published: October 12, 2023*
+
+*By: [Maximilian Lombardo](mailto:mlombardo@chanzuckerberg.com) and [Pablo Garcia-Nieto](pgarcia-nieto@chanzuckerberg.com)*
+
+The Census team is thrilled to announce the introduction of a library-size normalized expression layer and pre-calculated cell and gene statistics. This work is reflected in the Census schema V1.1.0 tailored to empower your single-cell research.
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:7]: add link to schema

Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:7]: style: don't love the hyperbole of "thrilled".

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+
+With these additions users can easily:
+
+- Expand their Census query filters to select genes or cells based on the new metadata included, for example selecting only cells with N number of genes expressed.
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:11]: remove "included"

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+- Export the expanded cell and gene metadata for downstream analysis.
+- Export a normalized expression data matrix for downstream analysis.
+
+These features are currently exclusive to the "latest" versions of the Census data release. We invite your feedback as you explore these novel functionalities.
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:15]: perhaps note they will be included in the next LTS?

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 21]
```diff
@@ -0,0 +1,225 @@
...
+
+Keep on reading to find out more about these features!
+
+## Description of new data added to Census
+
+All of the following changes were introduced in the Census schema V1.1.0
+
+### Added a new library-size normalized layer
+
+We have introduced a library-size normalized X layer for the RNA measurements of both the human and mouse experiments under `X["normalized"]`.  The normalized layer is built by dividing each value in the raw count matrix by its corresponding row sum (i.e. size normalization).
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:25]: "under" is awkward.  Perhaps "available as X['normalized']" or "saved as..."?

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 22]
```diff
@@ -0,0 +1,225 @@
...
+
+It is important to note that in order to preserve high precision for the result of a division we would need to encode data with many decimal point values. This can lead to a representation of data that is very large on disk and on memory. As means to compress this matrix we introduced changes that lead to reduced numerical precision and slight deviations from the expected row sum of 1.0.
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:27]: this paragraph is super confusing (accurate, but hard to parse).

Perhaps something like: 
To reduce data size and improve performance, normalized values are stored with a reduced floating point precision. In addition, to ensure that small count values do not round to zero, a small sigma has been added. You will see the effect of these artifacts in row (per-cell) values not summing to precisely 1.0.

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 23]
```diff
@@ -0,0 +1,225 @@
...
+
+In addition, to avoid that non-zero values become zero values during the normalization due to the reduced numerical precision, a small sigma value is added during the calculation of the normalized layer.
+
+### Enhanced gene metadata
+
+The `ms["RNA"].var` data frame for both the human and mouse experiments has been enriched with two new metadata fields:
+
+- `nnz` the number of non-zero (nnz) values, effectively the number of cells expressing this gene.
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:35]: the list needs punctuation between the field and field description, e.g., "nnz: the number ..."

Comment (bkmartinjr) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:35]: oh, I notice it exists in a few places, and is just not consistently used.

Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:35]: IIRC, this is wrong -- `nzz` is the number of explicitly stored values, not the number of non-zero values.

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 24]
```diff
@@ -0,0 +1,225 @@
...
+- `n_measured_obs` the "measured" cells for this gene, effectively the number of cells for which this gene was measured in their respective dataset.
+
+### Enhanced cell metadata
+
+The `obs` data frame for both the human and mouse experiments is now augmented with the following new metadata, allowing users to forego common calculations used in early data pre-processing. For each cell:
+
+- `raw_sum` the count sum derived from `X["raw"]`.
+- `nnz`: the number of non-zero (nnz) values, effectively the number of genes expressed on this cell.
+- `raw_mean` the average counts.
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:46]: these names appear to be wrong - they should be raw_mean_nnz, etc.

Comment (bkmartinjr) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:46]: And given the subtlety of the denominator (ie.. mean of explicitly stored values), it may be worth calling that out, or suggesting "for more details, see the schema"

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 25]
```diff
@@ -0,0 +1,225 @@
...
+- `raw_variance` the variance of the counts.
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:47]: name is wrong - missing _nnz

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md (hunk 24 above)
Comment (pablo-gar) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:46]: vestigial, this article started back in August. Will correct

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 26]
```diff
@@ -0,0 +1,225 @@
...
+   obs_value_filter = cell_filter,
+   obs_column_names = cell_columns
+)
+```
+
+## Help us improve these data additions
+
+We encourage you to engage with these new features in the Census API and share your feedback. This input is invaluable for the ongoing enhancement of the Census project.
+
+For further information on the new library-size normalized layer, please reach out to us at [soma@chanzuckerberg.com](soma@chanzuckerberg.com). To report issues or for additional feedback, refer to our [Census GitHub repository](https://github.com/chanzuckerberg/cellxgene-census/issues).
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:227]: what about "further information" on the other features?   Suggest:   "For further information on any new feature, please reach out ...."

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 27]
```diff
@@ -0,0 +1,227 @@
...
```
Comment (bkmartinjr) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:227]: also, any links to notebooks, doc pages, etc?

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md (hunk 23 above)
Comment (pablo-gar) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:37]: I can correct the definition, but the sentence is "effectively the number of cells expressing this gene" is still correct, as Census does not store explicit store 0s, right?

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md (hunk 27 above)
Comment (pablo-gar) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:227]: Good point, will add some to the introduction.

Comment (pablo-gar) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:227]: Actually we don't have specific notebooks or docs around these new data additions

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md (hunk 23 above)
Comment (bkmartinjr) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md]: Yes.  But given how SOMA defines nnz, I thought it worth being explicit.  I leave it to you to decide :-)

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 28]
```diff
@@ -0,0 +1,225 @@
+# Introducing a normalized layer and pre-calculated cell and gene statistics in Census
+
+<!-- markdownlint-disable MD036 -->
+*Published: October 12, 2023*
+
+*By: [Maximilian Lombardo](mlombardo@chanzuckerberg.com) and [Pablo Garcia-Nieto](pgarcia-nieto@chanzuckerberg.com)*
+<!-- markdownlint-enable MD036 -->
+
+The Census team is happy to announce the introduction of a library-size normalized expression layer and pre-calculated cell and gene statistics. This work is reflected in changes introduced in the [Census schema V1.1.0](https://github.com/chanzuckerberg/cellxgene-census/blob/main/docs/cellxgene_census_schema.md) tailored to empower your single-cell research.
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:9]:
```suggestion
The Census team is happy to announce the introduction of two new data features, tailored to empower your single-cell research: a library-size, normalized expression layer and pre-calculated cell and gene statistics. This work is reflected in changes introduced in the [Census schema V1.1.0](https://github.com/chanzuckerberg/cellxgene-census/blob/main/docs/cellxgene_census_schema.md)```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 29]
```diff
@@ -0,0 +1,225 @@
...
+
+With these additions users can easily:
+
+- Expand their Census query filters to select genes or cells based on the new metadata, for example selecting only cells with N number of genes expressed.
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:13]:
```suggestion
- Expand their Census query filters to select genes or cells based on the new metadata. For example, selecting only cells with N number of genes expressed.
```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md (hunk 28 above)
Comment (atolopko-czi) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:9]: Also, what does "library-size" mean? Full Census?

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 30]
```diff
@@ -0,0 +1,225 @@
...
+
+### Added a new library-size normalized layer
+
+We have introduced a library-size normalized X layer for the RNA measurements of both the human and mouse experiments available as `X["normalized"]`.  The normalized layer is built by dividing each value in the raw count matrix by its corresponding row sum (i.e. size normalization).
+
+To reduce data size and improve performance, normalized values are stored with a reduced floating point precision. In addition, to ensure that small count values do not round to zero, a small sigma has been added. You will see the effect of these artifacts in row (per-cell) values not summing to precisely 1.0.
+
+### Enhanced gene metadata
+
+The `ms["RNA"].var` data frame for both the human and mouse experiments has been enriched with two new metadata fields:
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:33]:
```suggestion
The `ms["RNA"].var` DataFrame for both the human and mouse experiments has been enriched with two new metadata fields:
```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 31]
```diff
@@ -0,0 +1,225 @@
...
+
+- `nnz` — the number of explicitly stored values, effectively the number of cells expressing this gene.
+- `n_measured_obs` — the "measured" cells for this gene, effectively the number of cells for which this gene was measured in their respective dataset.
+
+### Enhanced cell metadata
+
+The `obs` data frame for both the human and mouse experiments is now augmented with the following new metadata, allowing users to forego common calculations used in early data pre-processing. For each cell:
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:40]:
```suggestion
The `obs` DataFrame for both the human and mouse experiments is now augmented with the following new metadata, allowing users to forego common calculations used in early data pre-processing. For each cell:
```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 32]
```diff
@@ -0,0 +1,225 @@
...
+
+- `raw_sum` — the count sum derived from `X["raw"]`.
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:42]:
```suggestion
- `raw_sum` — the sum of the raw counts, derived from `X["raw"]`.
```

Comment (atolopko-czi) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:42]: I find this rewording clearer, but YMMV

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 33]
```diff
@@ -0,0 +1,225 @@
...
+- `nnz` — the number of explicitly stored values, effectively the number of genes expressed on this cell.
+- `raw_mean_nnz` — the average counts from explicitly stored values.
+- `raw_variance_nnz` — the variance of the counts from explicitly stored values.
+- `n_measured_vars` — the "measured" genes, effectively the number of genes measured in the dataset from which the cell was originated from, thus all cells from the same dataset have the same value for this variable.
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:46]:
```suggestion
- `n_measured_vars` — the "measured" genes, effectively the number of genes measured in the dataset from which the cell originated, thus all cells from the same dataset have the same value for this variable.
```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 34]
```diff
@@ -0,0 +1,225 @@
...
+import cellxgene_census
+
+with cellxgene_census.open_soma(census_version = "latest") as census
+    adata = cellxgene_census.get_anndata
+        census = census,
+        organism = "Homo sapiens",
+        var_value_filter = "feature_id in ['ENSG00000161798', 'ENSG00000188229']",
+        obs_value_filter = "cell_type == 'sympathetic neuron'",
+        column_names = {"obs": ["tissue", "cell_type"]},
+        X_name = "normalized" # Specificy the normalized layer for this query
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:64]:
```suggestion
        X_name = "normalized" # Specify the normalized layer for this query
```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 35]
```diff
@@ -0,0 +1,225 @@
...
+    )
+```
+
+Similarly, in R we can export the data to Seurat or SingleCellExperiment objects with the argument `X_layers` of the functions `get_seurat()` and `get_singel_cell_experiment()`.
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:68]:
```suggestion
Similarly, in R we can export the data to Seurat or SingleCellExperiment objects with the argument `X_layers` of the functions `get_seurat()` and `get_single_cell_experiment()`.
```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 36]
```diff
@@ -0,0 +1,225 @@
...
+           value_filter = "tissue == 'brain' and sex == 'male'"
+       )
+    )
+
+    # Set iterable for normalized matrix
+    iterator = query.X("normalized").tables()
+    
+    # Iterate over the normalized matrix.
+    # Get an iterative slice as pyarrow.Table
+    raw_slice = next (iterator)
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:133]:
```suggestion
    raw_slice = next(iterator)
```

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md [hunk 37]
```diff
@@ -0,0 +1,225 @@
...
+raw_slice <-  iterator$read_next()
+
+# Perform analysis
+```
+
+### Utilizing pre-calculated stats for querying `obs` and `var`
+
+To filter cells or genes based on pre-calculated statistics and export to AnnData, you can use the new metadata variables as value filters.
+
+For example you can you can add a filter to query cells with more than 500 genes expressed in addition to other filters. In Python this looks like the following.
```
Comment (atolopko-czi) [inline docs/articles/2023/20231012-normalized_layer_precalc_stats.md:169]:
```suggestion
For example, you can add a filter to query cells with more than 500 genes expressed, along with other filters. In Python this looks like the following.
```

**Review by atolopko-czi — DISMISSED** [review]
Some minor fixes and wording suggestions.

File: docs/articles/2023/20231012-normalized_layer_precalc_stats.md (hunk 28 above)
Comment (pablo-gar) [reply docs/articles/2023/20231012-normalized_layer_precalc_stats.md:9]: sum of counts in cell, it's common single-cell jargon

**Review by pablo-gar — APPROVED** [review]

**Review by pablo-gar — APPROVED** [review]
